### PR TITLE
feat: add CLI for management

### PR DESCRIPTION
**Pull request by steven-murray**
This makes an installable `hera_pipelines` package, whose main purpose is to provide a convenient locus for pipeline-management scripts. The benefit of doing it this way is that the scripts are then callable from anyway. I've used the recommended method for installing scripts so that there's one main entry point: `herapipe` and many sub-commands.

Currently I've only translated all the scripts in `utils/` and the notebook-readme script. I will add some of my own once the basic skeleton in this PR is accepted.

### CODE REVIEW

**Comment by jsdillon** [comment]
@aewallwi Since this impinges on your work, I'd also like to get your opinion.

**Comment by aewallwi** [comment]
I'm concerned about this approach's effect on the current workflow because we have been developing pipelines itteratively where we put nearly identical scripts and tomls in different folders. This has the advantage of minimizing the refactoring overhead when we want to try a small change -- we just copy the pipe we want to tweak to a new dir, make the tweaks to the toml, and run it in a different appropriate labeled directory as well. It seems like making this package installable will require all scripts for different pipeline versions to have different names to avoid conflicts which could introduce considerable overhead.

**Comment by jsdillon** [comment]
Could your concerns be alleviated by not moving anything out of utils (and not making any of that installable) and just opening an issue with the idea of graduating temporary scripts up to "installable" level?  

I certainly don't want to change the modality here where people developing push right to main here.

**Comment by aewallwi** [comment]
I haven't read this PR I'm detail so let me know if I'm not characterizing things correctly here. I am concerned about what I think is this approach's effect on the current. Workflow because we have been developing pipelines itteratively where we put nearly identical scripts and tomls in different folders. This has the advantage of minimizing the refactoring overhead when we want to try a small change. It seems like making this package installable will require all scripts for different pipeline versions to have different names to avoid conflicts.

**Comment by jsdillon** [comment]
I think the idea is that a small number of scripts (related to pipeline management) will be installable while most things (like DO scripts) will not be installed.

**Comment by aewallwi** [comment]
As long as this doesn't install the do scripts than I think things will be
fine. What directories/  pipeline utilities will be installed?

**Comment by jsdillon** [comment]
As it stands, it seems like Steven wants to convert the things in `utils` into installable scripts. Doesn't have to be that way though.

**Comment by steven-murray** [comment]
Hi @aewallwi, yes the DO scripts will stay as they are. The new installable portion is for "management" scripts that make life easier for the person running the pipeline. I have already written a few of these in getting H6C going, and it makes sense to keep them in a central location. As a for-instance, I converted everything in the `utils` folder into installable scripts (I can also keep them in `utils/` for some time if needed as a hold-over). 

Another for-instance would be my new script that helps with chugging through days of observation, staging them on-the-fly, building makeflows for them, running them makeflows, then unstaging.

File: .pre-commit-config.yaml
Comment (jsdillon) [inline .pre-commit-config.yaml:1]: This means the only thing getting checked is the installable stuff, right? Nothing in `pipelines` has any style enforced on it, right?

File: src/hera_pipelines/cli.py
Comment (jsdillon) [inline src/hera_pipelines/cli.py:2]: Would like to get @aewallwi's opinion on whether moving his scripts to here will break any of the ongoing pipeline work. If they will, then I don't think it's worth it to remove stuff from utils.

File: src/hera_pipelines/repo_management.py
Comment (jsdillon) [inline src/hera_pipelines/repo_management.py:6]: We probably want this to be a script that can be run from the command line.

**Review by jsdillon — COMMENTED** [review]
[edit: nevermind, it installs now]

File: .pre-commit-config.yaml
Comment (steven-murray) [reply .pre-commit-config.yaml:1]: Correct, I didn't want to go down that rabbit hole

File: src/hera_pipelines/cli.py
Comment (steven-murray) [reply src/hera_pipelines/cli.py:2]: I agree

File: src/hera_pipelines/repo_management.py
Comment (steven-murray) [reply src/hera_pipelines/repo_management.py:6]: it can be, with `hp notebook-readme <directory>`

Comment (jsdillon) [reply src/hera_pipelines/repo_management.py:6]: ah, I see... because that's a function in cli.py

**Comment by steven-murray** [comment]
@jsdillon  / @aewallwi is there anything blocking this from being merged?

I think the right way forward is for @aewallwi to add back any scripts currently being used in `utils/` to the `utils/` folder, probably with a top-level deprecation warning so that over time the pipelines/scripts that use those scripts can be converted over to use the new system. There's no rush in deleting them, we just want to make sure that eventually everyone's using the same thing.

**Review by jsdillon — APPROVED** [review]
I'm fine with this as long as @aewallwi is too

**Review by aewallwi — APPROVED** [review]
This looks good to me.

**Pull request merged**
